plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,326 @@
+	for _, v := range list {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *NFSClient) parseStat(mountpoint string, export string, version string, line []string, fullstat bool, acc telegraf.Accumulator) error {
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convert(line)
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var fields = make(map[string]interface{})
+
+	if fullstat && first == "events" && len(nline) >= len(eventsFields) {
+		for i, t := range eventsFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_events", fields, tags)
+	} else if fullstat && first == "bytes" && len(nline) >= len(bytesFields) {
+		for i, t := range bytesFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_bytes", fields, tags)
+	} else if fullstat && first == "xprt" && len(line) > 1 {
+		switch line[1] {
+		case "tcp":
+			if len(nline)+2 >= len(xprttcpFields) {
+				for i, t := range xprttcpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprttcp", fields, tags)
+			}
+		case "udp":
+			if len(nline)+2 >= len(xprtudpFields) {
+				for i, t := range xprtudpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprtudp", fields, tags)
+			}
+		}
+	} else if version == "3" || version == "4" {
+		if in(nfs3Fields, first) && len(nline) > 7 {
+			if first == "READ" {
+				fields["read_ops"] = nline[0]
+				fields["read_retrans"] = (nline[1] - nline[0])
+				fields["read_bytes"] = (nline[3] + nline[4])
+				fields["read_rtt"] = nline[6]
+				fields["read_exe"] = nline[7]
+				acc.AddFields("nfsstat_read", fields, tags)
+			} else if first == "WRITE" {
+				fields["write_ops"] = nline[0]
+				fields["write_retrans"] = (nline[1] - nline[0])
+				fields["write_bytes"] = (nline[3] + nline[4])
+				fields["write_rtt"] = nline[6]
+				fields["write_exe"] = nline[7]
+				acc.AddFields("nfsstat_write", fields, tags)
+			}
+		}
+		if fullstat && version == "3" {
+			if in(nfs3Fields, first) && len(nline) <= len(nfsopFields) {
+				for i, t := range nline {
+					item := fmt.Sprintf("%s_%s", first, nfsopFields[i])
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		} else if fullstat && version == "4" {
+			if in(nfs4Fields, first) && len(nline) <= len(nfsopFields) {
+				for i, t := range nline {
+					item := fmt.Sprintf("%s_%s", first, nfsopFields[i])
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var device string
+	var version string
+	var export string
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+		if in(line, "fstype") && (in(line, "nfs") || in(line, "nfs4")) && len(line) > 4 {
+			device = line[4]
+			export = line[1]
+		} else if (in(line, "(nfs)") || in(line, "(nfs4)")) && len(line) > 5 {
+			version = strings.Split(line[5], "/")[1]
```

Comment:
panic in the split()?  I think that unlikely, since NFS mounts are versioned, and the line in question looks something like this:
```
        RPC iostats version: 1.1  p/v: 100003/4 (nfs)
```
Which comes from `<linux_kernel>/net/sunrpc/stats.c` line 257-259:
```C
257         seq_printf(seq, "\tRPC iostats version: %s  ", RPC_IOSTATS_VERS);
258         seq_printf(seq, "p/v: %u/%u (%s)\n",
259                         clnt->cl_prog, clnt->cl_vers, clnt->cl_program->name);
```

Reply:
Why not just check the return of Split?

Reply:
I don't think this is resolved as you don't check the return. We are not in the kernel and that format might change in the future. Better safe than sorry!